Assets/Scripts/movement.cs

```diff
@@ -22,6 +28,69 @@ void Start()
     // Update is called once per frame
     void Update()
     {
-        
+        if(!PauseMenu.isPaused){
+        direction = new Vector2(Input.GetAxis("Horizontal"),Input.GetAxis("Vertical")).normalized;
+
+        body.velocity = direction * walkSpeed;
+
+        // handle directions
+        HandleSpriteFlip();
+
+        SetSprite();
+        }
+
+    }
+
+    void SetSprite() {
+        List<Sprite> directionSprites = GetSpriteDirection();
+        if(directionSprites != null) {
+            // holding dir
+            float playTime = Time.time - idleTime; // how long have we been moving
+            int frame = (int)((playTime * frameRate) % directionSprites.Count); // total frames since we started moving
+            spriteRenderer.sprite = directionSprites[frame];
+        } else {
+            // holding nothing
+            idleTime = Time.time;
+        }
+    }
+
+    void HandleSpriteFlip() {
```

Comment:
Vertical movement is handled by south and north sprites, i.e there is no need to flip the sprites because they already exist. This approach is done assuming that we only have east sprites, which is technically not true, but we can save memory by not having west sprites.